mevislab.github.io/content/tutorials/thirdparty/pytorch/pytorchexample2.md

```diff
@@ -0,0 +1,177 @@
+---
+title: "Example 2: TBD"
+date: 2023-06-30
+status: "OK"
+draft: false
+weight: 873
+tags: ["Advanced", "Tutorial", "PyTorch", "Python", "PythonPip", "AI"]
+menu: 
+  main:
+    identifier: "pytorchexample2"
+    title: "TBD"
+    weight: 873
+    parent: "pytorch"
+---
+# Example 2: TBD
+
+## Introduction
+In this example, you are using a pre-trained PyTorch deep learning model (HighRes3DNet) to perform a full brain parcellation. HighRes3DNet is a 3D residual network presented by Li et al. in [On the Compactness, Efficiency, and Representation of 3D Convolutional Networks: Brain Parcellation as a Pretext Task](https://link.springer.com/chapter/10.1007/978-3-319-59050-9_28).
+
+## Steps to do
+Add a `LocalImage` module to your workspace and select the file *MRI_Head.dcm*. For PyTorch it is necessary to resample the data to a defined size. Add a `Resample3D` module to the `LocalImage` and open the panel. Change *Keep Constant* to *Voxel Size* and define *Image Size* as 176, 217, 160.
+
+![Resample3D module](/images/tutorials/thirdparty/pytorch_example2_1.png "Resample3D module").
+
+The coordinates in PyTorch are also a little different than in MeVisLab, therefore you have to rotate the image. Add an `OrthoSwapFlip` module and connect it to the `Resample3D` module. Change *View* to *Other* and set *Orientation* to *YXZ*. Also check *Flip horizontal*, *Flip vertical* and *Flip depth*. *Apply* your changes.
+
+![OrthoSwapFlip module](/images/tutorials/thirdparty/pytorch_example2_2.png "OrthoSwapFlip module").
+
+You can use the Output Inspector to see the changes on the images after applying the resample and a swap or flip.
+
+{{< imagegallery 3 "images/tutorials/thirdparty/" "Original" "Resample3D" "OrthoSwapFlip">}}
+
+Add an `OrthoView2D` to your network and save the *\*.mlab* file.
+
+![OrthoView2D module](/images/tutorials/thirdparty/pytorch_example2_3.png "OrthoView2D module").
+
+## Integrate PyTorch and scripting
+For integrating PyTorch and Python scripting, we need a `PythonImage` module. Add it to your workspace. Right-click {{< mousebutton "right" >}} on the `PythonImage` module and select {{< menuitem "Grouping" "Add to new Group...">}}. Right-click {{< mousebutton "right" >}} your new group and select {{< menuitem "Grouping" "Add to new Group...">}}. Name your new local macro *DemoAI*, select a directory for your project and leave all settings as default.
+
+Our new module does not provide an input or output. 
+
+![DemoAI local macro](/images/tutorials/thirdparty/pytorch_example2_4.png "DemoAI local macro").
+
+### Add Interfaces to your local macro
+Right-click {{< mousebutton "right" >}} the local macro and select {{< menuitem "Related Files" "DemoAI.script">}}. MATE opens showing the *\*.script* file of our module. Add an input *Field* of type *Image*, an output *Field* using the *internalName* of the output of our `PythonImage` and a *Trigger* to start the segmentation.
```

Comment:
Rather "Adding an interface to the local macro" or "Providing the local macro with an interface"